software/controller/lib/core/controller.cpp

```diff
@@ -19,52 +19,92 @@ limitations under the License.
 
 static constexpr Duration LoopPeriod = milliseconds(10);
 
+// Controller gains, as inputs - set from external debug program, read but never
+// modified by controller.
+static DebugFloat dbg_blower_valve_ki("blower_valve_ki",
+                                      "Integral gain for blower valve PID",
+                                      VarAccess::ReadWrite, -1.0f);
+static DebugFloat dbg_blower_valve_kp("blower_valve_kp",
+                                      "Proportional gain for blower valve PID",
+                                      VarAccess::ReadWrite, 0.04f);
+static DebugFloat dbg_blower_valve_kd("blower_valve_kd",
+                                      "Derivative gain for blower valve PID");
```

Comment:
Variable access for DebugFloats are ReadWrite by default, and I only point this out because dbg_blower_valve_kd doesn't have the ReadWrite access explicitly written out like the other two. I it should be written out explicitly for clarity and consistency. Same goes for the decimal formatting.

Reply:
Also, why are these ReadWrite? Doesn't the comment above say it should be ReadOnly?

Reply:
Good catch on the lack of explicit access declaration. Maybe we should make them all explicit. I'll see if that is possible

ReadWrite refers to whether the debug interface (i.e. the user, via serial connection) can arbitrarily modify these as settings. The controller itself has r/w access to everything, bypassing this access control. The comment above explains that this is a one way street - user sets these. The controller side simply has no need to ever modify them internally.